enterprise/api/v1/serializers.py

```diff
@@ -410,3 +411,147 @@ class Meta:
         )
 
     enterprise_customer = EnterpriseCustomerSerializer()
+
+
+# pylint: disable=abstract-method
+class EnterpriseCustomerCourseEnrollmentsSerializer(serializers.Serializer):
+    """Serializes enrollment information for a collection of students/emails.
+
+    This is mainly useful for implementing validation when performing enrollment operations.
+    """
+    lms_user_id = serializers.CharField(required=False)
+    tpa_user_id = serializers.CharField(required=False)
+    user_email = serializers.EmailField(required=False)
+    course_run_id = serializers.CharField(required=True)
+    course_mode = serializers.ChoiceField(
+        choices=(
+            ('audit', 'audit'),
+            ('verified', 'verified'),
+            ('professional', 'professional')
+        ),
+        required=True
+    )
+    email_students = serializers.BooleanField(default=False, required=False)
+
+    enterprise_customer_user = None
+
+    def save(self, **kwargs):
+        """
+        Perform the enrollment for existing enterprise customer users, or create the pending objects for new users.
+        """
+        enterprise_customer = self.context.get('enterprise_customer')
+        user_email = self.validated_data.get('user_email')
+        course_run_id = self.validated_data.get('course_run_id')
+        course_mode = self.validated_data.get('course_mode')
+        email_students = self.validated_data.get('email_students')
+
+        if self.enterprise_customer_user:
+            self.enterprise_customer_user.enroll(course_run_id, course_mode)
+        else:
+            self.enterprise_customer_user = enterprise_customer.enroll_user_pending_registration(
+                user_email,
+                course_mode,
+                course_run_id
+            )
+
+        if email_students:
+            enterprise_customer.notify_enrolled_learners(
+                self.context.get('request_user'),
+                course_run_id,
+                [self.enterprise_customer_user]
+            )
```

Comment:
Do we want to notify if this was just an upgrade?

Reply:
I don't have a strong opinion, but I remember our conversation around Guild being that we would notify learners when they were upgraded. In any case, they still have the option to not notify learners if they know ahead of time they are performing an upgrade.